File: 3304/kazachkova_anna/lab1/lab1.py
#!/usr/bin/env python
#license removed
import rospy
import random
from geometry_msgs.msg import Twist

def lab1():
	publisher = rospy.Publisher('/turtle1/cmd_vel', Twist, queue_size=10)
	rospy.init_node('lab1', anonymous=True)
	rate = rospy.Rate(1)
	while not rospy.is_shutdown():
		vel_msg = Twist()
		vel_msg.linear.x = random.randint(0,3)
		vel_msg.linear.y= random.randint(0,3)
		vel_msg.linear.z=random.randint(0,3)
		vel_msg.angular.x = random.randint(0,3)
		vel_msg.angular.y = random.randint(0,3)
		vel_msg.angular.z = random.randint(0,2)
		rospy.loginfo(vel_msg)
		publisher.publish(vel_msg)
		rate.sleep()
if __name__ == '__main__':
	try:
		lab1()
	except rospy.ROSInterruptException:
	        pass

File: 3303/Faber_Vadim/Lab1/src/lab1.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include <math.h>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "lab1");
	ros::NodeHandle nh;
	ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 1000);

	ros::Rate rate(1);

	for (int t=0; t<8; t++){
		geometry_msgs::Twist msg;
		msg.linear.x = 2.5 * (t/2);
		msg.angular.z = (M_PI/2)*2;

		ROS_INFO_STREAM("\nlinear.x=" << msg.linear.x << "\n"
			<< "angular.z=" << msg.angular.z);
		pub.publish(msg);
		rate.sleep();
	}
	ros::spinOnce();
	return 0;
}

